xbmc/addons/kodi-addon-dev-kit/include/kodi/gui/ListItem.h

```diff
@@ -22,22 +22,44 @@
 #include "../AddonBase.h"
 #include "definitions.h"
 
+#include <memory>
+
 namespace kodi
 {
 namespace gui
 {
 
-  class CAddonGUIControlBase
+  class CWindow;
+
+  class CAddonGUIControlBase : public std::enable_shared_from_this<CAddonGUIControlBase>
```

Comment:
Becomes removed, has tested here to use them, but found on class constructor not usable and brings a `std::bad_weak_ptr` exception.